ftw/book/tests/test_table_export_import.py

```diff
@@ -29,6 +34,37 @@ def test_export_table(self, browser):
             'text/csv; charset=utf-8',
             browser.headers.get('content-type'))
 
+    @browsing
+    def test_non_ascii_export(self, browser):
+        table2 = create(Builder('table')
+                        .titled(u'Population')
+                        .with_table((('Ranking', 'City', 'Population'),
+                                     ('0', u'Gu\xe4ngzhou', '44 mil <sup>1</sup>')))
+                        .having(border_layout='grid',
+                                footnote_text=RichTextValue(
+                                    u'<p><sup>1</sup> thats quite big</p>'))
+                        .within(self.table.aq_parent))
+        browser.login().visit(table2, view='table_export_import')
+        try:
+            browser.click_on('Export')
+        except HTTPServerError:
+            self.fail('Export should work non ascii chars in the table')
+
+    @browsing
+    def test_bom_encoded_import(self, browser):
+        csvfile = (u'http://nohost/plone/the-example-book/historical-background/china/population;;\n\r'
+                   u'Ranking;City;Population\r\n'
+                   u'1;Gu\xe4ngzhou;100 mil').encode('utf-8-sig')  # Encode as utf-8 with BOM
+
+        browser.login().visit(self.table, view='table_export_import')
+        browser.fill({'Modified CSV-file:': (csvfile, 'file.csv', 'text/csv'),
+                      'Columns to import': 'City (Spalte 2)'})
+
+        browser.click_on('Import')
+        self.assertNotIn('Error The file does not seem to belong to this table.',
```

Comment:
This should be a positive test, not just a negative one IMO
- otherwise any change to the error message in [table_export_import.py#L120](https://github.com/4teamwork/ftw.book/blob/4c6c6e0dac86fe756c857bccc7db17f8f7ba5fe2/ftw/book/browser/table_export_import.py#L120) would make this test worthless.:-/

Test rather for "Successfully imported the column". (you could also try testing that `browser.css('.error').normalized_text()` is empty, but `.error ` might need to be more qualified for that.

Ta muchly!

Reply:
Döne